kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs

```diff
@@ -317,6 +317,64 @@ impl<'tcx> GotocHook<'tcx> for SliceFromRawPart {
     }
 }
 
+/// This hook intercepts calls to `memcmp` and skips CBMC's pointer checks if the number of bytes to be compared is zero.
+/// See issue https://github.com/model-checking/kani/issues/1489
+pub struct MemCmp;
+
+impl<'tcx> GotocHook<'tcx> for MemCmp {
+    fn hook_applies(&self, tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> bool {
+        let name = with_no_trimmed_paths!(tcx.def_path_str(instance.def_id()));
+        name.contains("memcmp")
+    }
+
+    fn handle(
+        &self,
+        tcx: &mut GotocCtx<'tcx>,
+        instance: Instance<'tcx>,
+        mut fargs: Vec<Expr>,
+        assign_to: Place<'tcx>,
+        target: Option<BasicBlock>,
+        span: Option<Span>,
+    ) -> Stmt {
+        let loc = tcx.codegen_span_option(span);
+        let target = target.unwrap();
+        let first = fargs.remove(0);
+        let second = fargs.remove(0);
+        let count = fargs.remove(0);
```

Comment:
`fargs` might be big for macro-generated code. You may consider doing something like the below link

https://stackoverflow.com/questions/43026118/what-is-the-idiomatic-way-to-remove-the-first-n-elements-in-a-mutable-vec

Reply:
Could also use `into_iter()` followed by 3 nexts.

Reply:
`fargs` should have exactly that set of arguments, no more no less

Reply:
Yes,`memcmp` takes exactly 3 arguments, so this shouldn't be an issue.